### ISSUE TITLE
Inconsistency between is_array and is_arr[]

### ISSUE THREAD

**Issue by Lexikos** [TheArkive/JXON_ahk2#4]
Some parts of the code use `is_array`, which is based on `type(obj)`, and others use `is_arr[obj]`, which is based on whether the object was created from "[" or "{". It would make sense to just remove the is_arr map and use type() in all instances (or better, use `obj is Array`).

**Comment by TheArkive** [comment]
Ok.  I'll admit that part of the code was very confusing to me at first, but I see it now.

I pushed a simplification that is more consistent in both functions.  Let me know if that will suffice.

**Comment by TheArkive** [comment]
@Lexikos 

Is the latest push sufficient?

**Comment by Lexikos** [comment]
Seems sufficient. The main thing was to eliminate the obsolete `is_arr` array.

One reason I suggest using `is Array` is that I am using a modified version of Jxon_Load which accepts the map and array classes as parameters. I am passing it a map subclass with __get to allow the use of property syntax.

**Comment by TheArkive** [comment]
Ok, I'll look into using `is Array` then.

I also plan to try and refactor this lib to see if I can squeeze any more performance out of it, but that project is a ways off.

**Comment by TheArkive** [comment]
@Lexikos 

Now using `obj is Array`.  Does this work for your uses?  I also added streamlining changes to `jxon_dump()`.

**Comment by Lexikos** [comment]
I think I intended to take a look and reply, but to be frank, I stopped caring about any of this shortly after opening the Issue. (I used the modified script to implement a quick fix for a private project, and wasn't remotely concerned about improving the code for my own use. I didn't even remove `is_arr`.)

It looks like Dump still relies on Type(), which would prevent the use of a custom Array/Map subclass. Apparently I just removed that check entirely. I've only used Dump for debugging so far (printing the result of Parse, which in my case uses a custom Map subclass, as I mentioned).
